go-update/src/main/java/org/reactome/release/goupdate/DuplicateReporter.java

```diff
@@ -81,23 +84,29 @@ public DuplicateReporter(MySQLAdaptor adaptor)
 			{
 				for (GKInstance i : instances)
 				{
-					for (GKSchemaAttribute referrerAttrib : (Set<GKSchemaAttribute>)i.getSchemClass().getReferers())
-					{
-						@SuppressWarnings("unchecked")
-						Collection<GKInstance> referrers = (Collection<GKInstance>) i.getReferers(referrerAttrib);
-						if (referrerCounts.containsKey(i.getDBID()))
-						{
-							referrerCounts.put(i.getDBID(), referrerCounts.get(i.getDBID()) + referrers.size());
-						}
-						else
-						{
-							referrerCounts.put(i.getDBID(), referrers.size());
-						}
-					}
+					Long dbid = i.getDBID();
+					int refCount = getReferrerCountforInstance(i, classesToIgnore);
+					referrerCounts.put(dbid,refCount);
 				}
 			}
 		}
 		return referrerCounts;
 	}
 	
+	public int getReferrerCountforInstance(GKInstance instance, String ... classesToIgnore) throws Exception
```

Comment:
It might be better to make this method private.  Also, if you like the use of the stream in `getReferrerCountForAccession` above it would be best to handle the exception in the body rather than throw it.

Reply:
Ah okay.  What do you think about using the stream in the comment for `getReferrerCountForAccession`?  If you like it, then the lambda would need to handle the exception thrown here or this method would need to handle the exception rather than throw it.